validphys2/src/validphys/config.py

```diff
@@ -1634,16 +1649,11 @@ def produce_filter_data(self, fakedata: bool = False, theorycovmatconfig=None):
         if not fakedata:
             return validphys.filters.filter_real_data
         else:
-            if theorycovmatconfig is not None and theorycovmatconfig.get(
-                "use_thcovmat_in_sampling"
-            ):
-                # NOTE: By the time we run theory covmat closure tests,
-                # hopefully the generation of pseudodata will be done in python.
-                raise ConfigError(
-                    "Generating closure test data which samples from the theory "
-                    "covariance matrix has not been implemented yet."
-                )
-            return validphys.filters.filter_closure_data_by_experiment
+            if theorycovmatconfig is not None:
```

Comment:
I guess the `theorycovmatconfig.get("use_thcovmat_in_sampling")` should still be checked.